lib/linter.js

```diff
@@ -70,30 +70,31 @@ const commentParser = new ConfigCommentParser();
  * @returns {void}
  */
 function addDeclaredGlobals(globalScope, configGlobals, commentDirectives) {
-    Object.keys(configGlobals).forEach(name => {
-        let variable = globalScope.set.get(name);
-
-        if (!variable) {
-            variable = new eslintScope.Variable(name, globalScope);
-            variable.eslintExplicitGlobal = false;
-            globalScope.variables.push(variable);
-            globalScope.set.set(name, variable);
-        }
-        variable.writeable = configGlobals[name];
-    });
-
-    Object.keys(commentDirectives.enabledGlobals).forEach(name => {
-        let variable = globalScope.set.get(name);
+    const allGlobals = Object.assign(
```

Comment:
Nit: I don't particularly like this variable name, as this is not just globals, but rather an object with globals as keys and an object with comment information and global config as a value. If nothing else, this might deserve a JSDoc type to clarify the expected object structure. I'll leave it to you to identify the best path forward here.

Reply:
Updated the variable name to hopefully prevent it from being misleading -- let me know if that's better.